### PR TITLE
adds support for CocoaPods and Swift PM

### PR DESCRIPTION
**Pull request by brennanMKE**
I've added the necessary support for CocoaPods and SwiftPM.

- [ ] Retain name of Promise on CocoaPods or change to a unique name
- [ ] Merge PR to master
- [ ] Run the following commands below successfully

```sh
git tag 2.0.0
git push --tags
bundle install
bundle exec pod spec lint
bundle exec pod trunk push
```

### CODE REVIEW

**Comment by brennanMKE** [comment]
Hey @khanlou, have you had time to chat with Orta about taking over the Promise pod?

**Comment by khanlou** [comment]
I did mention it to him, but didn't get a response. I'll try a different member of the cocoapods team.

**Comment by khanlou** [comment]
@brennanMKE okay so it looks like we can't take over old cocoapods names. i think we'd need to come up with a new name for it. any ideas?

**Comment by samsonjs** [comment]
- Oath
- Pinky
- Honor
- Eventually
- Later
- PromiseNoPeopleThisIsTheGoodOne ... 🤔

**Comment by SintraWorks** [comment]
Is “Promises” not an option?

**Comment by khanlou** [comment]
"Promises" is taken: https://github.com/CocoaPods/Specs/blob/d0ec5a65e80656c8d78e12ff19f251df879e0bc2/Specs/1/5/5/Promises/0.1.0/Promises.podspec.json

Tempted to run with `Promise.v3.final-FINAL.ai`

**Comment by brennanMKE** [comment]
I see `Promise` is now deprecated on CocoaPods. Perhaps the original owner would now be willing to transfer control of the name for this implementation. cc @khanlou 

https://cocoapods.org/pods/Promise

**Comment by khanlou** [comment]
@brennanMKE working on it: https://twitter.com/orta/status/855871899682918400

**Comment by randomiser** [comment]
Any progress with this btw? Would be great to be able to include this dependency easily.

**Comment by brennanMKE** [comment]
I'll message the owner of the Promises lib. I know him.

**Comment by joshdholtz** [comment]
@khanlou Sending you a message in iOS Folks slack

**Comment by joshdholtz** [comment]
We should be good here now! Donated my `Promises` Cocoapods namespace to this cause ✊ 

RIP https://github.com/joshdholtz/ios-promises 💀

**Comment by randomiser** [comment]
Awesome. Thanks very much.

File: .swift-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.0
```
Comment (khanlou) [inline .swift-version:1]: should this be 3.1 now?

File: Promise.podspec [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Pod::Spec.new do |s|
+  s.name             = "Promise"
```
Comment (khanlou) [inline Promise.podspec:2]: can you change the to `Promises`?

File: Promise.podspec [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  s.version          = "2.0.0"
+  s.summary          = "A Promise library for Swift"
+  s.description      = "A Promise library for Swift, based partially on Javascript's A+ spec"
+  s.module_name      = "Promise"
+  s.homepage         = "https://github.com/khanlou/Promise"
+  s.license          = 'MIT'
+  s.author           = { "Soroush Khanlou" => "soroush@khanlou.com" }
+  s.source           = { :git => "https://github.com/khanlou/Promise.git", :tag => "2.0.0" }
```
Comment (khanlou) [inline Promise.podspec:10]: can you update the tag to `v2.0`

File: README.md [hunk 4]
```diff
@@ -18,16 +18,52 @@ Using a special type to represent values that will exist in the future means tha
...
+Once you have your Swift package set up, adding AlamPromiseofire as a dependency is as easy as adding it to the `dependencies` value of your `Package.swift`.
 
 ```swift
-let usersPromise = fetchUsers() // Promise<[User]>
-usersPromise.then({ users in
-    self.users = users
-})
+dependencies: [
+    .Package(url: "https://github.com/khanlou/Promise.git", majorVersion: 2)
+]
```
Comment (khanlou) [inline README.md:53]: i think installation instructions should go at the end of the README. ime people have more problems understanding how they work than how to install a cocoapod. can you please update the cocoapod name here as well?

**Review by khanlou — COMMENTED** [review]
ok, now that we have the `Promises` name on cocoapods, i added a tag for `v2.0` (and made some breaking changes i've been meaning to make 😈 ). @brennanMKE can you make these changes and then i'll merge and push up to pod trunk?

**Comment by brennanMKE** [comment]
I will get to this when I have time. It may not be till this weekend.

**Pull request merged**